apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -411,6 +412,55 @@ export const publishResource = async (
   return addedVersionResult
 }
 
+export const getAncestryWithSelf = async ({
+  siteId,
+  resourceId,
+}: {
+  siteId: number
+  resourceId: number
+}) => {
+  const ancestorsWithSelf = await db
+    .withRecursive("Resources", (eb) =>
+      eb
+        .selectFrom("Resource")
+        .select([
+          "Resource.id",
+          "Resource.title",
+          "Resource.permalink",
+          "Resource.parentId",
+          "Resource.type",
+        ])
+        .where("Resource.siteId", "=", Number(siteId))
+        .where("Resource.id", "=", String(resourceId))
+        .where((eb) =>
+          eb.and([eb("Resource.type", "!=", ResourceType.RootPage)]),
+        )
```

Comment:
this `AND` clause seems redundant since there is only one conditional. Can be simplified?

```suggestion
        .where("Resource.type", "!=", ResourceType.RootPage),
```